### PR TITLE
Destructors should not throw exceptions

### PR DESCRIPTION
**Pull request by yisonPylkita**
Issue discussion #1

### CODE REVIEW

File: inotify/InotifyManager.cpp [hunk 1]
```diff
@@ -20,13 +20,6 @@ InotifyManager::~InotifyManager(){
 		delete it->second;
 	}
 	watchSet.clear();
-
-	/* Close inotify fd */
-	if(fd != -1 && close(this->fd)){
```
Comment (tmick0) [inline inotify/InotifyManager.cpp:25]: I think we still ought to attempt to close this fd, and maybe emit a warning if that fails

Comment (Bu11etmagnet) [inline inotify/InotifyManager.cpp:25]: This leaks the fd (it is never closed).

**Review by Bu11etmagnet — CHANGES_REQUESTED** [review]
The fd should be closed even if we don't throw.

File: inotify/InotifyManager.cpp (hunk 1 above)
Comment (yisonPylkita) [reply inotify/InotifyManager.cpp:25]: Thanks for noticing. Fixed

**Comment by yisonPylkita** [comment]
@Bu11etmagnet Should be fine now, right ?

**Comment by Bu11etmagnet** [comment]
@yisonPylkita I can't see anything wrong with it.

**Comment by yisonPylkita** [comment]
@le1ca Can you merge this now ?

**Review by tmick0 — APPROVED** [review]

**Comment by tmick0** [comment]
Thanks for the fixes everyone, it's merged now

**Pull request merged**